tests/Unit/Domain/Test_Orientations.cpp

```diff
@@ -0,0 +1,181 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
```

Comment:
Okay, we need to come to a decision; since I have sympathy with both arguments, I went and looked at the [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md).  For concrete classes, the recommendation is that if you provide operator= then you should provide operator== (see C.10 and C.11).  So I think for these simple classes we should follow this and provide the operartors == and !=.